poc_iot_verifier/src/loader.rs

```diff
@@ -157,34 +162,57 @@ impl Loader {
                     LoraBeaconIngestReportV1::decode(buf)?.try_into()?;
                 tracing::debug!("beacon report from ingestor: {:?}", &beacon);
                 let packet_data = beacon.report.data.clone();
-                tracing::debug!("beacon data: {:?}", &packet_data);
-                Report::insert_into(
-                    &self.pool,
-                    beacon.ingest_id(),
-                    beacon.report.remote_entropy,
-                    packet_data,
-                    buf.to_vec(),
-                    &beacon.received_timestamp,
-                    ReportType::Beacon,
-                )
-                .await
+                match self.check_valid_gateway(&beacon.report.pub_key).await {
```

Comment:
nit: Arguably this logic can be hoisted in a function like `insert_report_for_existing_gateway`? Just so we don't do too much nesting.

Reply:
yeah maybe but I am not sure I like putting too much logic into the `DB` modules, ie in this case poc_report.rs